network/gossip/libp2p/libp2pNode.go

```diff
@@ -215,8 +215,6 @@ func (p *P2PNode) AddPeer(ctx context.Context, peer NodeAddress) error {
 		return fmt.Errorf("failed to add peer %s: %w", peer.Name, err)
 	}
 
-	p.Lock()
```

Comment:
This lock will be held till the connect of line 220 does not return. The `Connect` internally does the backoff for 5s, 6s, 9s,14s etc. if the remote node is not responding.

Reply:
The lock is not needed. The `libp2p.Connect` is concurrency safe